### DISCUSSION TITLE
Hub admin panel import (avatar or scene) to custom reticulum throws error

### DISCUSSION THREAD

**Discussion by willnnotdan** [mozilla/hubs#4708] [Hubs Cloud Questions] [answered]
Hi - Ive been setting up each of the hubs components seperately (reticulum, hubs, spoke) in Azure and Ive managed to stand most of it up and get them to communicate but Ive run into a problem. 

The problem is when I want to upload content to the server via the admin panel. Ive got the cors sorted so now I enter the URL and Im presented with the robots to import, however when I import it throws an error which looks like it may be failing to write it to the database, but I may be wrong about that

This is what you get in the browser when importing

<img width="1792" alt="Screenshot 2021-09-22 at 11 56 29" src="https://user-images.githubusercontent.com/14215427/136007185-793d0dd1-295d-43ee-9e62-8c74b96e9ff4.png">

And this is the output in reticulums console
`[debug] QUERY OK source="oauth_providers" db=0.2ms idle=99.4ms
SELECT o0."oauth_provider_id", o0."source", o0."provider_account_id", o0."provider_access_token", o0."provider_access_token_secret", o0."account_id", o0."inserted_at", o0."updated_at", o0."account_id" FROM "ret0"."oauth_providers" AS o0 WHERE (o0."account_id" = $1) ORDER BY o0."account_id" [1061319624315895810]
[info] Sent 500 in 34ms
[error] #PID<0.16660.0> running RetWeb.Endpoint (connection #PID<0.15988.0>, stream id 43) terminated
Server: hubs.local:4000 (https)
Request: GET /api/postgrest/avatars?avatar_sid=ilike.%2AcmooWx4%2A&order=id.desc
** (exit) an exception was raised:
    ** (ArgumentError) argument error: [[], {:jose_jws, {:jose_jws_alg_rsa_pkcs1_v1_5, :RS512}, :undefined, %{"typ" => "JWT"}}, %{"account_id" => 1061319624315895810, "aud" => :ret_perms, "create_hub" => true, "exp" => 1632308445, "iat" => 1632308145, "iss" => "", "jti" => "94851c46-c657-43fa-9833-95b8c041afa9", "nbf" => 1632308144, "postgrest_role" => :ret_admin, "sub" => "1061319624315895810_global", "tweet" => false, "typ" => "access"}]
        (jose) src/jwt/jose_jwt.erl:184: :jose_jwt.sign/3
        (guardian) lib/guardian/token/jwt.ex:259: Guardian.Token.Jwt.create_token/3
        (guardian) lib/guardian.ex:765: Guardian.returning_tuple/1
        (guardian) lib/guardian.ex:594: Guardian.encode_and_sign/4
        (ret) lib/ret/perms_token.ex:24: Ret.PermsToken.token_for_perms/1
        (ret) lib/ret_web/plugs/rewrite_authorization_header_to_perms.ex:15: RetWeb.Plugs.RewriteAuthorizationHeaderToPerms.call/2
        (ret) RetWeb.Router.proxy_api/2
        (ret) lib/ret_web/router.ex:1: RetWeb.Router.__pipe_through1__/1
        (phoenix) lib/phoenix/router.ex:283: Phoenix.Router.__call__/2
        (ret) lib/plug/error_handler.ex:65: RetWeb.Router.call/2
        (ret) lib/ret_web/endpoint.ex:1: RetWeb.Endpoint.plug_builder_call/2
        (ret) lib/plug/debugger.ex:132: RetWeb.Endpoint."call (overridable 3)"/2
        (ret) lib/ret_web/endpoint.ex:1: RetWeb.Endpoint."call (overridable 4)"/2
        (ret) lib/ret_web/endpoint.ex:1: RetWeb.Endpoint.call/2
        (phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:42: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy) /home/moz/reticulum/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
        (cowboy) /home/moz/reticulum/deps/cowboy/src/cowboy_stream_h.erl:296: :cowboy_stream_h.execute/3
        (cowboy) /home/moz/reticulum/deps/cowboy/src/cowboy_stream_h.erl:274: :cowboy_stream_h.request_process/3
        (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3`

**Comment by willnnotdan** [comment]
Any ideas?

**Reply by pattersonbl2** [reply]
looks like the ret server is having issues with a get request from the Postgres database.  We currently don't have support for azure deployments so I am unsure what the error could be.

**Reply by willnnotdan** [reply]
Okay thanks

**Comment by willnnotdan** [comment, marked as answer]
For anyone that comes across this issue it is resolved by following these instructions but specifically not the copy option 
https://github.com/mozilla/hubs-ops/wiki/Running-PostgREST-locally
